### DISCUSSION TITLE
Emacs not showing Unicode characters on welcome screen

### DISCUSSION THREAD

**Discussion by voidtrance** [syl20bnr/spacemacs#16102] [General]
If I run emacs without a filename on the command line, it starts and opens a welcome screen, which lists the recently opened files. However, it seems to fail to display unicode characters on that screen:
![image](https://github.com/syl20bnr/spacemacs/assets/30448940/6ff4e9d5-3c9f-4b50-9f1b-66236d3af1ff)

The default font set in my config is Source Code Pro and I have the `unicode-layer` installed.

I would appreciate any help in fixing this.

**Comment by henriquenj** [comment]
Is your Emacs opening on the terminal? If yes then it's your terminal font that needs to support the symbols in question. What is defined in the Emacs config is not relevant. How to change the font on your terminal depends on what kind of terminal emulator you are using (Windows Terminal, Gnome terminal, terminator, etc).

I believe what is missing in your font are the programming symbols. A popular place to get fonts with those is https://www.nerdfonts.com/

**Reply by voidtrance** [reply]
My Emacs is opening in a separate window and I've confirmed that the font setting in my `.spacemacs` file does take effect. I also downloaded and installed several of the fonts from the link that you posted (Thank you!). However, it is still not showing the symbols.

Some of the symbols did show up but it's missing the symbols for C and C++ files. Although, when I open a C file, the crumbs on the top of the window are showing the C file symbol but are not showing other ones:
![image](https://github.com/syl20bnr/spacemacs/assets/30448940/c140030d-f490-46a8-9f75-20ebdc918a25)
(The above is with DejaVuSansM Nerd Font.)

**Reply by henriquenj** [reply]
And when you open Emacs in another way i.e. not calling from the terminal, then you don't see the problem?

**Reply by voidtrance** [reply]
No matter how I start the process - from clicking a desktop icon or from typing the command line in a terminal - my emacs always starts as a separate windowed process. Basically, I don't use the `-nw` parameter.

Either way I start emacs, the icons don't show up. When I start emacs with `-nw`, it doesn't even try to render those lines on the welcome screen.

**Reply by henriquenj** [reply]
Very weird. I speculate that it must be something related with the desktop environment. I guess it should affect other programs as well.

You can inspect these variables in your Emacs to see if you have any weird flag in your build `system-configuration-features system-configuration-options system-configuration`

**Reply by henriquenj** [reply]
Maybe you are just missing `all-the-icons` fonts https://github.com/domtronn/all-the-icons.el#installing-fonts Run `SPC SPC all-the-icons-install-fonts`

**Reply by voidtrance** [reply]
That was it. Thanks.
